frontend/catalyst/jax_tracer.py

```diff
@@ -624,6 +624,7 @@ def trace_quantum_operations(
     ctx: JaxTracingContext,
     trace: DynamicJaxprTrace,
     mcm_config: qml.devices.MCMConfig = qml.devices.MCMConfig(),
+    static_compile: bool = False,
```

Comment:
Let's not make this an option. Instead if the parameters are static, produce the static version of the IR.

Reply:
But if you want, we can keep this as an option and then remove it. It may be easier to have it as an option for debugging and testing.

Reply:
I assumed that we don't necessarily want to have the "straight line" static compilation to be the default behaviour. That's why I added the option, but yeah if we want to remove, it It would be better to remove it at the end.